src/Processors/Transforms/Streaming/AggregatingTransform.cpp

```diff
@@ -498,7 +498,7 @@ void AggregatingTransform::checkpoint(CheckpointContextPtr ckpt_ctx)
         }
 
         /// Serializing no shared data
-        params->aggregator.checkpoint(variants, wb);
+        DB::serialize(variants, wb, params->aggregator);
```

Comment:
Does variants already points to params->aggregator at this point ?

Reply:
if there is no data, variants points to nullptr aggregator.